src/analysis/locally_convex/disk.lean

```diff
@@ -0,0 +1,189 @@
+
+lemma nhds_basis_abs_convex : (𝓝 (0 : E)).has_basis
+  (λ (s : set E), s ∈ 𝓝 (0 : E) ∧ balanced 𝕜 s ∧ convex ℝ s) id :=
+begin
+  refine (locally_convex_space.convex_basis_zero ℝ E).to_has_basis (λ s hs, _)
+    (λ s hs, ⟨s, ⟨hs.1, hs.2.2⟩, rfl.subset⟩),
+  refine ⟨convex_hull ℝ (balanced_core 𝕜 s), _, convex_hull_min (balanced_core_subset s) hs.2⟩,
+  refine ⟨filter.mem_of_superset (balanced_core_mem_nhds_zero hs.1) (subset_convex_hull ℝ _), _⟩,
+  refine ⟨balanced_convex_hull_of_balanced (balanced_core_balanced s), _⟩,
+  exact convex_convex_hull ℝ (balanced_core 𝕜 s),
+end
+
+variables [has_continuous_smul ℝ E] [topological_add_group E]
+
+lemma nhds_basis_abs_convex_open : (𝓝 (0 : E)).has_basis
+  (λ (s : set E), (0 : E) ∈ s ∧ is_open s ∧ balanced 𝕜 s ∧ convex ℝ s) id :=
+begin
+  refine (nhds_basis_abs_convex 𝕜 E).to_has_basis _ _,
+  { rintros s ⟨hs_nhds, hs_balanced, hs_convex⟩,
+    refine ⟨interior s, _, interior_subset⟩,
+    exact ⟨mem_interior_iff_mem_nhds.mpr hs_nhds, is_open_interior,
+      hs_balanced.interior (mem_interior_iff_mem_nhds.mpr hs_nhds), hs_convex.interior⟩ },
+  rintros s ⟨hs_zero, hs_open, hs_balanced, hs_convex⟩,
+  exact ⟨s, ⟨hs_open.mem_nhds hs_zero, hs_balanced, hs_convex⟩, rfl.subset⟩,
+end
+
+end nontrivially_normed_field
+
+section absolutely_convex_sets
+
+variables [topological_space E] [add_comm_monoid E] [has_zero E] [semi_normed_ring 𝕜]
+variables [has_smul 𝕜 E] [has_smul ℝ E]
+
+variables (𝕜 E)
+
+/-- The type of absolutely convex open sets. -/
+def abs_convex_open_sets :=
+{ s : set E // (0 : E) ∈ s ∧ is_open s ∧ balanced 𝕜 s ∧ convex ℝ s }
+
+instance abs_convex_open_sets.has_coe : has_coe (abs_convex_open_sets 𝕜 E) (set E) := ⟨subtype.val⟩
+
+namespace abs_convex_open_sets
+
+variables {𝕜 E}
+
+lemma coe_zero_mem (s : abs_convex_open_sets 𝕜 E) : (0 : E) ∈ (s : set E) := s.2.1
+
+lemma coe_is_open (s : abs_convex_open_sets 𝕜 E) : is_open (s : set E) := s.2.2.1
+
+lemma coe_nhds (s : abs_convex_open_sets 𝕜 E) : (s : set E) ∈ 𝓝 (0 : E) :=
+s.coe_is_open.mem_nhds s.coe_zero_mem
+
+lemma coe_balanced (s : abs_convex_open_sets 𝕜 E) : balanced 𝕜 (s : set E) := s.2.2.2.1
+
+lemma coe_convex (s : abs_convex_open_sets 𝕜 E) : convex ℝ (s : set E) := s.2.2.2.2
+
+end abs_convex_open_sets
+
+instance : nonempty (abs_convex_open_sets 𝕜 E) :=
+begin
+  rw ←exists_true_iff_nonempty,
+  dunfold abs_convex_open_sets,
+  rw subtype.exists,
+  exact ⟨set.univ, ⟨mem_univ 0, is_open_univ, balanced_univ, convex_univ⟩, trivial⟩,
+end
+
+end absolutely_convex_sets
+
+variables [is_R_or_C 𝕜]
+variables [add_comm_group E] [topological_space E]
+variables [module 𝕜 E] [module ℝ E] [is_scalar_tower ℝ 𝕜 E]
+variables [has_continuous_smul ℝ E]
+
+variables (𝕜 E)
+
+/-- The family of seminorms defined by the gauges of absolute convex open sets. -/
+noncomputable
+def maximal_seminorm_family : seminorm_family 𝕜 E (abs_convex_open_sets 𝕜 E) :=
+λ s, gauge_seminorm s.coe_balanced s.coe_convex (absorbent_nhds_zero s.coe_nhds)
```

Comment:
I'm not a big fan of this name, to me `maximal_seminorm_family` should refer to the set of all continuous seminorms (which also generates the topology iirc), not to only the gauges. My instinct would be `gauge_seminorm_family`.

Reply:
I think this gives you all seminorms (if you have a continuous seminorm, then it's unit ball is a absolutely convex open set), but `gauge_seminorm_family` is a way better name.